src/js/button.js

```diff
@@ -7,13 +12,13 @@
  * @class
  * @constructor
  */
-vjs.Button = vjs.Component.extend({
+var Button = Component.extend({
   /**
    * @constructor
    * @inheritDoc
    */
   init: function(player, options){
```

Comment:
I really like the plugin idea, that would keep the core code cleaner. When you update a major version you need to take a look to know what has changed and fix it. If by adding one extra plugin/script you can make make your code temporary work. It gives you the opportunity to start migrating all your code to es6 on separates steps instead of all at once.